### ISSUE TITLE
officially support android jetpack credential manager

### ISSUE THREAD

**Issue by wildsylvan** [flutter/flutter#154205]
### Use case

[Google Sign-In for Android is deprecated and will be removed from the Google Play Services Auth SDK. (com.google.android.gms:play-services-auth) in 2025](https://developer.android.com/identity/sign-in/legacy-gsi-migration)

### Proposal

there's already a community [package](https://pub.dev/packages/credential_manager) but having an official one would be better, since it's such core concern

**Comment by Sliman-Baghouri** [comment]
Yes now the only way to do it is via custom method channels or using 3rd party community plugin, it would be nice to have an official source

**Comment by stuartmorgan** [comment]
Based on [this blog post](https://android-developers.googleblog.com/2024/09/streamlining-android-authentication-credential-manager-replaces-legacy-apis.html) Google Sign-In for Android will be removed in H2 2025 specifically.
- I am hoping to publish a wrapper plugin in early (Q1) 2025, with the goal of having it be a replacement backend for the `google_sign_in` plugin (although we may publish a direct wrapper separately as well). This is likely to be paired with significant changes to the `google_sign_in` APIs, since every platform now has a different recommended underlying, so the current API may not be a good match (see https://github.com/flutter/flutter/issues/119300)
- We will definitely have something ready in advance of H2 2025.

**Comment by xlg** [comment]
It seems quite urgent.

**Comment by vinxzhu** [comment]
any update?

**Comment by Azbesciak** [comment]
Yes, would be good to have anything. Fast moving is good, but not in removing things, Google...

**Comment by stuartmorgan** [comment]
This is still planned for Q1.

**Comment by stuartmorgan** [comment]
https://github.com/flutter/flutter/issues/119300#issuecomment-2748917680 has a link to the design document that will be part of this transition. Unfortunately, after investigation, it is not possible to use `CredentialManager` to build a fully compatible replacement for the deprecated SDK, so just replacing the backend of the current API as a short-term transition isn't viable. (A semi-compatible shim is possible, but not one that could be released as a non-breaking change.)

As a note on adoption timelines, while Google Sign-In for Android will be removed from new versions of the `com.google.android.gms:play-services-auth` later this year, as noted in the link in the issue description, the service itself will continue to work past that point. This means that existing versions of the `google_sign_in` plugin will continue to work past the SDK removal point, so long as applications using it don't have other dependencies on new versions of `com.google.android.gms:play-services-auth`, so the SDK-level removal should not be a hard deadline on most clients adopting the new version of `google_sign_in` with its breaking changes.

The expectation is still that we will have the new version of `google_sign_in` available in advance of the removal from the SDK, however, so that adoption can begin before even edge cases could see compilation issues.

**Comment by fan123199** [comment]
Is there any further status update? Q1 ended nearly a month ago. While the existing sign-in works, the new credential manager significantly shortens the sign-in flow on Android. I really don't want the sign-in experience in Flutter is worse than native Android.

**Comment by ycv005** [comment]
Waiting for this.
